src/lib.rs

```diff
@@ -277,30 +277,41 @@ pub struct Metadata {
     pub max_depth: usize,
 }
 
+/// Helper trait for `core::iter::Peekable`.
+pub trait Peekable: core::iter::Iterator {
+    fn peek(&mut self) -> core::option::Option<&Self::Item>;
+}
+
+impl<I: core::iter::Iterator> Peekable for core::iter::Peekable<I> {
+    fn peek(&mut self) -> core::option::Option<&Self::Item> {
+        core::iter::Peekable::peek(self)
+    }
+}
+
 /// Derive-able trait for structures that can be mutated using serialized paths and values.
 pub trait Miniconf {
     /// Update an element by path.
     ///
     /// # Args
-    /// * `path` - The path to the element.
+    /// * `path` - The path to the element. With '/' as separator.
     /// * `data` - The serialized data making up the content.
     ///
     /// # Returns
     /// The result of the configuration operation.
-    fn set(&mut self, path: &str, data: &[u8]) -> Result<(), Error> {
-        self.set_path(path.split('/').peekable(), data)
+    fn set(&mut self, path: &str, data: &[u8]) -> Result<usize, Error> {
+        self.set_path(&mut path.split('/').peekable(), data)
     }
 
     /// Retrieve a serialized value by path.
     ///
     /// # Args
-    /// * `path` - The path to the element.
+    /// * `path` - The path to the element. With '/' as separator.
     /// * `data` - The buffer to serialize the data into.
     ///
     /// # Returns
     /// The number of bytes used in the `data` buffer for serialization.
     fn get(&self, path: &str, data: &mut [u8]) -> Result<usize, Error> {
-        self.get_path(path.split('/').peekable(), data)
+        self.get_path(&mut path.split('/').peekable(), data)
```

Comment:
Why pass a mutable reference instead of handing ownership for the iterator through the chain? There's no reason we need to retain ownership here.

Reply:
Within `get()` no.
But for `get_path()` the iterator may be expensive to create and we don't need to own it. Since there is no need to own it and ownership being more restrictive than a mutable borrow while not being faster, the mutable borrow is preferrable IMO.